packages/cells/src/cells/dropdown-cell.tsx

```diff
@@ -0,0 +1,97 @@
+import { CustomCell, ProvideEditorCallback } from "@glideapps/glide-data-grid";
+import styled from "styled-components";
+import * as React from "react";
+import Select, { MenuProps, components } from "react-select";
+import { CustomCellRenderer } from "../types";
+
+interface CustomMenuProps extends MenuProps<any> {}
+
+export class CustomMenu extends React.Component<CustomMenuProps> {
+    render() {
+        const { Menu } = components;
+        const { children, ...props } = this.props;
+        return <Menu {...props}>{children}</Menu>;
+    }
+}
+
+interface DropdownCellProps {
+    readonly kind: "dropdown-cell";
+    readonly value: string;
+    readonly allowedValues: readonly string[];
+    readonly readonly?: boolean;
+}
+
+export type DropdownCell = CustomCell<DropdownCellProps>;
+
+const Wrap = styled.div`
+    display: flex;
+    flex-direction: column;
+    align-items: stretch;
+`;
+
+const editor: ProvideEditorCallback<DropdownCell> = () => {
```

Comment:
I'm pretty sure you can do
```ts
const InnerEditor: ReturnType<ProvideEditorCallback<DropdownCell>> = p => {
    // ...
}

InnerEditor.disablePadding = true

const editor: ProvideEditorCallback<DropdownCell> = () => InnerEditor;
```

This way `editor` will now return a stable reference all the time, and you won't have to ignore `rules-of-hooks`

Reply:
You are right, fixed